tutorials/Signing_and_notarization_on_macOS/README.md

```diff
@@ -0,0 +1,177 @@
+    * Choose the `Developer ID Application` certificate type.
+    * Upload your Certificate Signing Request from the previous step.
+    * Download and install the certificate (drag & drop the certificate into the `Keychain Access` application).
+
+#### Viewing installed certificates
+
+You can find all installed certificates and their keychains by running the following command:
+```
+/usr/bin/security find-certificate -c "Developer ID Application"
+```
+
+If you have multiple `Developer ID Application` certificates installed,
+you will need to specify the path to the keychain, containing
+the certificate intended for signing.
+
+## Preparing an App ID
+
+#### Viewing existing App IDs
+
+Open https://developer.apple.com/account/resources/identifiers/list
+
+#### Creating a new App ID
+
+1. Open https://developer.apple.com/account/resources/identifiers/add/bundleId
+2. Choose `App ID` option.
+3. Choose `App` type.
+4. Enter a unique id to the `Bundle ID` field. 
+   It is recommended to use the reverse DNS notation for your domain (e.g.
+   `com.yoursitename.yourappname`).
+   
+## Creating an app-specific password
+
+To be able to upload an app for notarization,
+you will need an app-specific password associated with your Apple ID.
+
+Follow these steps to generate a new password:
+1. Sign in to your [Apple ID](https://appleid.apple.com/account/home) account page.
+2. In the Security section, click Generate Password below App-Specific Passwords.
+
+See [this Apple support page](https://support.apple.com/en-us/HT204397) for more information
+on the app-specific passwords.
+
+## Adding an app-specific password to a keychain
+
+To avoid remembering your one-time password, or writing it in scripts,
+you can add it to the keychain by running:
+```
+# Any name can be used instead of NOTARIZATION_PASSWORD
+
+xcrun altool --store-password-in-keychain-item "NOTARIZATION_PASSWORD"
+             --username <apple_id>
+             --password <password>
+```
+
+Then you'll be able to refer to the password like `@keychain:NOTARIZATION_PASSWORD`
+without the need to write the password itself.
+
+## Configuring Gradle
+
+All properties are set in `macOS` DSL block of Compose Desktop DSL:
+```kotlin
+import org.jetbrains.compose.compose
+import org.jetbrains.compose.desktop.application.dsl.TargetFormat
+
+plugins {
+    kotlin("jvm")
+    id("org.jetbrains.compose")
+}
+
+dependencies {
+    implementation(compose.desktop.currentOS)
+}
+
+compose.desktop {
+    application {
+        mainClass = "example.MainKt"
+
+        nativeDistributions {
+            targetFormats(TargetFormat.Dmg)
+
+            macOS {
+                bundleID = "com.example-company.example-app"
+            
+                signing { 
+                   identity = "John Doe"
+                }
+                notarization {
+                   appleID = "john.doe@example-company.com"
+                   password = "@keychain:NOTARIZATION_PASSWORD"
```

Comment:
Suggest people adding password and Apple ID to buildscript doesn't look like a good idea. Consider getting those from Gradle properties.